### PR TITLE
Bump i18n-js to v4

### PR DESCRIPTION
**Pull request by oliverguenther**
I18n.js v4 has a number of breaking changes:

- The configuration format changed, outputting the translations to json by default
- The rails middleware is removed
- The export is now a command instead of a rake task
- The javascript library is now an NPM package.

https://github.com/fnando/i18n-js/blob/main/MIGRATING_FROM_V3_TO_V4.md

https://community.openproject.org/work_packages/45926

### CODE REVIEW

File: frontend/src/app/shared/components/grids/widgets/time-entries/current-user/time-entries-current-user.component.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export class WidgetTimeEntriesCurrentUserComponent extends AbstractWidgetCompone
     const duration = this.entries.reduce((current, entry) => current + this.timezone.toHours(entry.hours), 0);
 
     if (duration > 0) {
-      return this.i18n.t('js.units.hour', { count: this.formatNumber(duration) });
```
Comment (ulferts) [inline frontend/src/app/shared/components/grids/widgets/time-entries/current-user/time-entries-current-user.component.ts:43]: We could use I18n's `numberToRounded` with a 2 digit precision to achieve the same as we had before.

File: lib/tasks/assets.rake [hunk 2]
```diff
@@ -78,5 +78,11 @@ namespace :assets do
   end
 
   desc 'Export frontend locale files'
-  task export_locales: ['i18n:js:export']
+  task :export_locales do
+    puts "Exporting I18n.js locales"
+    time = Benchmark.realtime do
+      I18nJS.call(config_file: Rails.root.join('config/i18n.yml'))
+    end
+    puts "=> Done in #{time.round(2)}s"
```
Comment (ulferts) [inline lib/tasks/assets.rake:86]: I would have expected this to be called somewhere for both the test as well as the production environment. How is this dealt with?

**Review by ulferts — CHANGES_REQUESTED** [review]
Big thanks for that bump @oliverguenther. 

I have two remarks, questions mostly.

File: frontend/src/app/shared/components/grids/widgets/time-entries/current-user/time-entries-current-user.component.ts (hunk 1 above)
Comment (oliverguenther) [reply frontend/src/app/shared/components/grids/widgets/time-entries/current-user/time-entries-current-user.component.ts:43]: Yes we could do that, this is merely a typescript error fix that is now apparent with i18n.js bringing its own types (we passed in a string before, which didn't correctly match the count so it was just returned).

File: lib/tasks/assets.rake (hunk 2 above)
Comment (oliverguenther) [reply lib/tasks/assets.rake:86]: This bit is actually unchanged. The rake task `assets:precompile` is extended to call this task, and just its contents have changed now.

Here's the enhance call: https://github.com/opf/openproject/blob/dev/lib/tasks/assets.rake#L33-L35

Comment (ulferts) [reply lib/tasks/assets.rake:86]: Ah, I didn't notice that the `export_locales` existed before and thus was already referenced in the build pipeline. Thanks for pointing it out.

**Pull request merged**